main.tf

```diff
@@ -104,7 +104,7 @@ data "aws_iam_policy_document" "main" {
     ]
 
     resources = [
-      "arn:${data.aws_partition.current.partition}:iam::*:mfa/&{aws:username}",
+      "arn:${data.aws_partition.current.partition}:iam::*:mfa/*",
```

Comment:
you can entirely remove the `mfa/*` line in here. the `iam:DeactivateMFADevice` permission is based on user:
check the DeactivateMFADevice row in here: (https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsidentityandaccessmanagementiam.html)